letta/services/agent_manager.py

```diff
@@ -407,6 +409,260 @@ def detach_block_with_label(
+                        combined_query.c.created_at > cursor_query
+                    )
+                else:
+                    main_query = main_query.where(
+                        combined_query.c.created_at < cursor_query
+                    )
+
+            # Add ordering
+            if not embed_query:  # Skip if already ordered by similarity
+                if ascending:
+                    main_query = main_query.order_by(
+                        combined_query.c.created_at.asc(),
+                        combined_query.c.id.asc(),
+                    )
+                else:
+                    main_query = main_query.order_by(
+                        combined_query.c.created_at.desc(),
+                        combined_query.c.id.asc(),
+                    )
+
+            # Add limit
+            if limit:
+                main_query = main_query.limit(limit)
+
+            # Execute query
+            results = list(session.execute(main_query))
+
+            passages = []
+            for row in results:
+                data = dict(row._mapping)
+                if data['agent_id'] is not None:
+                    # This is an AgentPassage - remove source fields
+                    data.pop('source_id', None)
+                    data.pop('file_id', None)
+                    passage = AgentPassage(**data)
+                else:
+                    # This is a SourcePassage - remove agent field
+                    data.pop('agent_id', None)
+                    passage = SourcePassage(**data)
+                passages.append(passage)
+            
+            return [p.to_pydantic() for p in passages]
+
+
+            # # Verify agent exists and user has permission to access it
+            # agent = AgentModel.read(db_session=session, identifier=agent_id, actor=actor)
+
+            # # Use the lazy-loaded relationships to get passages
+            # # Sort by created_at
+            # all_passages = sorted(
+            #     [*agent.source_passages, *agent.agent_passages],
+            #     key=lambda x: x.created_at,
+            #     reverse=(not ascending)
+            # )
+
+
+            # return [passage.to_pydantic() for passage in all_passages]
+
+    @enforce_types
+    def passage_size(
+        self, 
+        actor           : PydanticUser,
+        agent_id        : Optional[str] = None,
+        file_id         : Optional[str] = None, 
+        limit           : Optional[int] = 50,
+        query_text      : Optional[str] = None,
+        start_date      : Optional[datetime] = None,
+        end_date        : Optional[datetime] = None,
+        cursor          : Optional[str] = None,
+        source_id       : Optional[str] = None,
+        embed_query     : bool = False,
+        ascending       : bool = True,
+        embedding_config: Optional[EmbeddingConfig] = None,
+        agent_only      : bool = False
+    ) -> int:
+        return len(
```

Comment:
rework this, limit should not be included here (doesn't make sense to call a size with a limit)

Reply:
should refactor the underlying queries with this and list_passage to be shared up until the limit